playback/src/audio_backend/rodio.rs

```diff
@@ -0,0 +1,96 @@
+use super::{Open, Sink};
+extern crate rodio;
+use std::{io, thread, time};
+use std::process::exit;
+
+pub struct RodioSink {
+    rodio_sink: rodio::Sink,
+}
+
+fn list_outputs() {
+    println!("Default Audio Device:\n  {:?}", rodio::default_output_device().map(|e| e.name()));
+
+    println!("Available Audio Devices:");
+    for device in rodio::output_devices() {
+        println!("- {}", device.name());
+        // Output formats
+        if let Ok(fmt) = device.default_output_format() {
+            println!("  Default format:\n    {:?}", fmt);
+        }
+        let mut output_formats = match device.supported_output_formats() {
+            Ok(f) => f.peekable(),
+            Err(e) => {
+                println!("Error: {:?}", e);
+                continue;
+            },
+        };
+        if output_formats.peek().is_some() {
+            println!("  All formats:");
+            for format in output_formats {
+                println!("    {:?}", format);
```

Comment:
What's with the whitespace in this `println` and others? is this just for prettifying stuff?